embassy-stm32/src/lib.rs

```diff
@@ -180,6 +180,18 @@ pub fn init(config: Config) -> Peripherals {
     }
 
     unsafe {
+        #[cfg(feature = "_split-pins-enabled")]
```

Comment:
@Dirbaio this is where it will fail anyway if the user enables a split feature for a wrong chip. Not sure whether we can mitigate this risk.